pkg/sensors/accelerator/acclerator.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accelerator
+
+//nolint:gci // The supported device imports are kept separate.
+import (
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/exp/slices"
+
+	"github.com/pkg/errors"
+	"github.com/sustainable-computing-io/kepler/pkg/sensors/accelerator/device"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	// Add supported devices.
+	_ "github.com/sustainable-computing-io/kepler/pkg/sensors/accelerator/device/sources"
+)
+
+var (
+	accelerators map[string]Accelerator
```

Comment:
Not a fan of globals

I would rather that you create a:

```go 
type AcceleratorRegistry struct {
   registry map[AcceleratorType]Accelerator
}
```

That would allow us to easily create various mock registry configurations without having to override package globals etc...

It can also be protected under mutex if we registrations might occur from different goroutines.